src/main/java/com/exactpro/th2/infraoperator/operator/manager/impl/DefaultWatchManager.java

```diff
@@ -514,6 +515,10 @@ public void eventReceived(Action action, Th2Link th2Link) {
 
                 int refreshedBoxCount = 0;
 
+                checkForSameName(th2Link.getSpec().getBoxesRelation().getRouterMq(), annotationFor(th2Link));
```

Comment:
Don't you expect NPE here?

Reply:
in Th2LinkSpec they are set to `new ArrayList<>()` if they are null